src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs

```diff
@@ -75,84 +77,114 @@ namespace RadixDlt.NetworkGateway.GatewayApi.Handlers;
 
 internal class DefaultEntityHandler : IEntityHandler
 {
-    private const int DefaultPageLimit = 100; // TODO make it configurable
-
     private readonly ILedgerStateQuerier _ledgerStateQuerier;
     private readonly IEntityStateQuerier _entityStateQuerier;
+    private readonly IOptionsMonitor<EndpointOptions> _endpointConfiguration;
 
-    public DefaultEntityHandler(ILedgerStateQuerier ledgerStateQuerier, IEntityStateQuerier entityStateQuerier)
+    public DefaultEntityHandler(ILedgerStateQuerier ledgerStateQuerier, IEntityStateQuerier entityStateQuerier, IOptionsMonitor<EndpointOptions> endpointConfiguration)
     {
         _ledgerStateQuerier = ledgerStateQuerier;
         _entityStateQuerier = entityStateQuerier;
-    }
-
-    public async Task<GatewayModel.EntityResourcesResponse?> Resources(GatewayModel.EntityResourcesRequest request, CancellationToken token = default)
-    {
-        var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
-        var aggregatePerVault = request.AggregationLevel == GatewayModel.ResourceAggregationLevel.Vault;
-
-        return await _entityStateQuerier.EntityResourcesSnapshot((GlobalAddress)request.Address, aggregatePerVault, ledgerState, token);
+        _endpointConfiguration = endpointConfiguration;
     }
 
     public async Task<GatewayModel.StateEntityDetailsResponse> Details(GatewayModel.StateEntityDetailsRequest request, CancellationToken token = default)
     {
         var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
+        var aggregatePerVault = request.AggregationLevel == GatewayModel.ResourceAggregationLevel.Vault;
 
         // currently we enforce exactly one element in request.Addresses
-        var item = await _entityStateQuerier.EntityDetailsItem((GlobalAddress)request.Addresses.Single(), ledgerState, token);
+        var item = await _entityStateQuerier.EntityDetailsItem((GlobalAddress)request.Addresses.Single(), aggregatePerVault, ledgerState, token);
 
         return new GatewayModel.StateEntityDetailsResponse(ledgerState, new List<GatewayModel.StateEntityDetailsResponseItem> { item });
     }
 
-    public async Task<GatewayModel.EntityMetadataResponse?> Metadata(GatewayModel.EntityMetadataRequest request, CancellationToken token = default)
+    public async Task<GatewayModel.StateEntityMetadataPageResponse?> Metadata(GatewayModel.StateEntityMetadataPageRequest request, CancellationToken token = default)
     {
-        var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
+        var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(
+            new GatewayModel.LedgerStateSelector { StateVersion = request.AtLedgerStateVersion }, token);
```

Comment:
I'm afraid we'll have to revert this idea. :( It was a good one but the moment we decided that those pagination endpoints could be used as entry points rather continuation points only we should get this fully-fledged `at_leder_state`.

_consider this open question that does not prevent merger_

Reply:
fixed.